### PR TITLE
Dev stage 3

### LINKED ISSUE #3: A minibatch support
Right now a neural network accepts only single flatten sample as an input.
Layer shape: (n_samples, )
Minibatch should be: (batch_size, n_samples, )

### LINKED ISSUE #8: Mnist loading
Add auutomatical downloading and unpacking mnist instead of local copy

### LINKED ISSUE #12: TQDM for cool training output

### PR DESCRIPTION
**Pull request by kuweg**

### CODE REVIEW

**Comment by kuweg** [comment]
TODO:
1. Fix getting a nan values using softmax with relu
2. Add example of solving regression problem on boston housing dataset
